paimon-core/src/main/java/org/apache/paimon/table/sink/TableWriteImpl.java

```diff
@@ -146,6 +156,18 @@ public void compact(BinaryRow partition, int bucket, boolean fullCompaction) thr
         write.compact(partition, bucket, fullCompaction);
     }
 
+    @Override
+    public TableWrite withMetricRegistry(MetricRegistry registry, BinaryRow partition, int bucket) {
+        write.withMetrics(
+                new CompactionMetrics(registry, tableName, getPartitionString(partition), bucket));
+        return this;
+    }
```

Comment:
One `TableWrite` (and also one `FileStoreWrite`) is responsible for writing all partitions and buckets of a table. You can't specify partitions and buckets here. Please check `AbstractFileStoreWrite` to understand how writers for different buckets are grouped together.

Reply:
Yes, I've notice that when developing the code for bridging to flink metrics system, I'll update this asap. Thank you for reminding me~ @tsreaper